contracts/governance/extensions/GovernorOverrideDelegateVote.sol

```diff
@@ -0,0 +1,194 @@
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        return quorum(proposalSnapshot(proposalId)) <= proposalVote.forVotes + proposalVote.abstainVotes;
+    }
+
+    /**
+     * @dev See {Governor-_voteSucceeded}. In this module, the forVotes must be strictly over the againstVotes.
+     */
+    function _voteSucceeded(uint256 proposalId) internal view virtual override returns (bool) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        return proposalVote.forVotes > proposalVote.againstVotes;
+    }
+
+    function _countVote(
+        uint256 proposalId,
+        address account,
+        uint8 support,
+        uint256 totalWeight,
+        bytes memory params
+    ) internal virtual override returns (uint256) {
+        if (keccak256(params) == keccak256(hex"23b70c8d0000000000000000000000000000000000000000")) {
+            return _countVotesOverride(proposalId, account, support, params);
+        }
+
+        totalWeight -= _overrideVoteWeight[account][proposalId];
+
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        if (proposalVote.voteReceipt[account].hasVoted) {
+            revert GovernorAlreadyCastVote(account);
+        }
+        proposalVote.voteReceipt[account] = VoteReceipt({hasVoted: true, support: support});
+
+        if (support == uint8(VoteType.Against)) {
+            proposalVote.againstVotes += totalWeight;
+        } else if (support == uint8(VoteType.For)) {
+            proposalVote.forVotes += totalWeight;
+        } else if (support == uint8(VoteType.Abstain)) {
+            proposalVote.abstainVotes += totalWeight;
+        } else {
+            revert GovernorInvalidVoteType();
+        }
+
+        return totalWeight;
+    }
+
+    function _countVotesOverride(
+        uint256 proposalId,
+        address account,
+        uint8 support,
+        bytes memory params
+    ) private returns (uint256) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+        uint256 proposalSnapshot = proposalSnapshot(proposalId);
+        address delegate = _getPastDelegate(account, proposalSnapshot);
+
+        if (proposalVote.overrideVoteReceipt[account].hasVoted) {
+            revert GovernorAlreadyCastVoteOverride(account);
+        }
+
+        uint256 overrideWeight = _getPastBalanceOf(account, proposalSnapshot);
+
+        proposalVote.overrideVoteReceipt[account] = VoteReceipt({hasVoted: true, support: support});
+        if (support == uint8(VoteType.Against)) {
+            proposalVote.againstVotes += overrideWeight;
+        } else if (support == uint8(VoteType.For)) {
+            proposalVote.forVotes += overrideWeight;
+        } else if (support == uint8(VoteType.Abstain)) {
+            proposalVote.abstainVotes += overrideWeight;
+        } else {
+            revert GovernorInvalidVoteType();
+        }
```

Comment:
This block is repeated a couple of times, I would try to encapsulate it somehow (may be in a private function that receives an `operator` function pointer, or the `proposalVote` might save votes in an array so that we only do `proposalVote[support] += ...`)

Reply:
Agree with this. I'll use a private function that tallies the votes.